angular/projects/admin/src/app/admin/components/dialog/dialog.component.ts

```diff
@@ -2,17 +2,21 @@ import { Component, OnInit, Inject } from '@angular/core';
 import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
 
 @Component({
-  selector: 'tanam-document-type-dialog-delete',
-  templateUrl: './document-type-dialog-delete.component.html',
-  styleUrls: ['./document-type-dialog-delete.component.scss']
+  selector: 'tanam-dialog',
+  templateUrl: './dialog.component.html',
+  styleUrls: ['./dialog.component.scss']
 })
-export class DocumentTypeDialogDeleteComponent {
+export class DialogComponent implements OnInit {
```

Comment:
Let's make an interface for the component that takes
 - `title: string`
 - `message: string`
 - `buttons: DialogButton[]`
 - `icon: DialogIcon`

Where `DialogButton` and `DialogIcon` is defined as such
```typescript
export type DialogButton = 'ok' | 'yes' | 'no' | 'cancel';
export type DialogIcon= 'warning' | 'info' | 'error';
```

See Android's dialog builder as an inspiration: https://developer.android.com/guide/topics/ui/dialogs#AlertDialog

Reply:
Resolve on this commit: https://github.com/oddbit/tanam/pull/115/commits/83e5b2f1baeae0016458c54f7094d881084c27cc